hs_core/hydroshare/resource.py

```diff
@@ -705,9 +705,10 @@ def delete_resource_file(pk, filename_or_id, user):
     else:
         raise ObjectDoesNotExist(filename_or_id)
 
-    if resource.raccess.public:
+    if resource.raccess.public or resource.raccess.discoverable:
         if not resource.can_be_public:
             resource.raccess.public = False
+            resource.raccess.discoverable = False
```

Comment:
yeah, that sounds good to me. On second thought, do we really need this 'can_be_public_or_discoverable" method in the first place? We can just use "has_all_required_elements()" and "has_required_content_files" when we need to check whether a resource is public or discoverable, right? This is also related to my comments below on calling the same checks twice to show informative messages.